hoodie-common/src/main/java/com/uber/hoodie/common/util/buffer/MemoryBoundedBuffer.java

```diff
@@ -30,25 +31,26 @@
 import java.util.function.Function;
 import org.apache.log4j.LogManager;
 import org.apache.log4j.Logger;
-import org.apache.spark.util.SizeEstimator;
 
 /**
  * Used for buffering input records. Buffer limit is controlled by {@link #bufferMemoryLimit}. It
  * internally samples every {@link #RECORD_SAMPLING_RATE}th record and adjusts number of records in
  * buffer accordingly. This is done to ensure that we don't OOM.
  *
+ * This buffer supports multiple producer single consumer pattern.
+ *
  * @param <I> input payload data type
  * @param <O> output payload data type
  */
-public class BufferedIterator<I, O> implements Iterator<O> {
+public class MemoryBoundedBuffer<I, O> implements Iterable<O> {
```

Comment:
Rename: BoundedInmemoryBuffer or should we just call it a Queue? given we are talking Producer and Consumer in the methods/classes around these